packages/component-test-utils-react/src/dispatcher/index.js

```diff
@@ -10,8 +9,30 @@ exports.createDispatcher = () => {
     useMemo: () => {},
     useReducer: () => {},
     useRef: () => {}, */
-    useState: initialState => {
-      return [initialState];
+
+  constructor(shallowedComponent) {
+    this.states = [];
+    this._firstCall = true;
+    this._shallowedComponent = shallowedComponent;
+  }
+
+  useState(initialState) {
```

Comment:
From what I see here you're globally rewriting the way `useState` is working under the hood so. Does it mean that `useState` is part of the real `Dispatcher` implmeentation in React?

Does it means that if you rewrite every React hooks for the shallow rendering purpose, you'll put them here?

If you have a link to the React implementation of the dispatcher, I think it would be a great idea to put it in comment, just to remember 😊

Reply:
useState is part of the real dispatcher yes. It made me 2 days of reading the react code source before i understand ^^

Yes, each react hooks will be in the dispatcher !

I will add the comment, great Idea <3